### PR TITLE
Update SDKs with new autorest.csharp version

### LINKED ISSUE #27297: Renaming any properties that use the name “Type” as this conflicts with the dotnet concept of a Type
There are a lot of properties simply named `Type`. We should rename them based on the type of the property if the type is not string. For instance, `public RuleType Type` can renamed to `public RuleType RuleType`.

### LINKED ISSUE #27325: Improve ResponseError
Remove ResponseError model if it's never used.
Otherwise, we want to see once more if we can make the deserialization work with the Azure.Core type.
At minimums we want to manually modify this type so that we don't have the extra layer with one property inside it.

### PR DESCRIPTION
**Pull request by fengzhou-msft**
The changes included are type renaming and removal of ErrorResponse model from common type.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by ArthurMa1978** [comment]
https://github.com/azure/azure-sdk-for-net/issues/27297

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/api/Azure.ResourceManager.DeviceUpdate.netstandard2.0.cs [hunk 1]
```diff
@@ -271,8 +271,8 @@ namespace Azure.ResourceManager.DeviceUpdate.Models
     public partial class CheckNameAvailabilityRequest
     {
         public CheckNameAvailabilityRequest() { }
+        public string CheckNameAvailabilityRequestType { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/api/Azure.ResourceManager.DeviceUpdate.netstandard2.0.cs:274]: This should be named ResourceType.

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/AzureFirewallApplicationRuleCollection.cs [hunk 2]
```diff
@@ -46,14 +46,14 @@ internal AzureFirewallApplicationRuleCollection(string id, string name, string e
         /// <summary> The action type of a rule collection. </summary>
         internal AzureFirewallRCAction Action { get; set; }
         /// <summary> The type of action. </summary>
-        public AzureFirewallRCActionType? ActionType
+        public AzureFirewallRCActionType? AzureFirewallRCActionType
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/AzureFirewallApplicationRuleCollection.cs:49]: So our rule is prepend `{ResourceName}` to any name which ends with `Type`?

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.Serialization.cs [hunk 3]
```diff
@@ -17,10 +17,10 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
             writer.WriteStartObject();
             writer.WritePropertyName("name");
             writer.WriteStringValue(Name);
-            if (Optional.IsDefined(Type))
+            if (Optional.IsDefined(EndpointType))
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorEndpoint.Serialization.cs:20]: Why is this `EndpointType`, not `ConnectionMonitorEndpointType`?

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorOutput.Serialization.cs [hunk 4]
```diff
@@ -15,10 +15,10 @@ public partial class ConnectionMonitorOutput : IUtf8JsonSerializable
         void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
         {
             writer.WriteStartObject();
-            if (Optional.IsDefined(Type))
+            if (Optional.IsDefined(OutputType))
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectionMonitorOutput.Serialization.cs:18]: same

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectivityIssue.cs [hunk 5]
```diff
@@ -22,13 +22,13 @@ internal ConnectivityIssue()
         /// <summary> Initializes a new instance of ConnectivityIssue. </summary>
         /// <param name="origin"> The origin of the issue. </param>
         /// <param name="severity"> The severity of the issue. </param>
-        /// <param name="type"> The type of issue. </param>
+        /// <param name="issueType"> The type of issue. </param>
         /// <param name="context"> Provides additional context on the issue. </param>
-        internal ConnectivityIssue(Origin? origin, Severity? severity, IssueType? type, IReadOnlyList<IDictionary<string, string>> context)
+        internal ConnectivityIssue(Origin? origin, Severity? severity, IssueType? issueType, IReadOnlyList<IDictionary<string, string>> context)
```
Comment (archerzz) [inline sdk/network/Azure.ResourceManager.Network/src/Generated/Models/ConnectivityIssue.cs:27]: same

File: sdk/network/Azure.ResourceManager.Network/src/Generated/Models/AzureFirewallApplicationRuleCollection.cs (hunk 2 above)
Comment (archerzz) [reply sdk/network/Azure.ResourceManager.Network/src/Generated/Models/AzureFirewallApplicationRuleCollection.cs:49]: Never mind. I think the rule is to name property name after its type name.

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs [hunk 6]
```diff
@@ -4571,8 +4571,8 @@ public enum CheckNameAvailabilityReason
     public partial class CheckNameAvailabilityRequest
     {
         public CheckNameAvailabilityRequest(string name) { }
+        public string CheckNameAvailabilityResourceType { get { throw null; } }
```
Comment (fengzhou-msft) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/api/Azure.ResourceManager.Sql.netstandard2.0.cs:4574]: ResourceType

File: sdk/webpubsub/Azure.ResourceManager.WebPubSub/api/Azure.ResourceManager.WebPubSub.netstandard2.0.cs [hunk 7]
```diff
@@ -305,9 +305,9 @@ public partial class NameAvailability
     }
     public partial class NameAvailabilityParameters
     {
-        public NameAvailabilityParameters(string type, string name) { }
+        public NameAvailabilityParameters(string nameAvailabilityParametersType, string name) { }
         public string Name { get { throw null; } }
-        public string Type { get { throw null; } }
+        public string NameAvailabilityParametersType { get { throw null; } }
```
Comment (fengzhou-msft) [inline sdk/webpubsub/Azure.ResourceManager.WebPubSub/api/Azure.ResourceManager.WebPubSub.netstandard2.0.cs:310]: ResourceType

File: sdk/cdn/Azure.ResourceManager.Cdn/src/Generated/Models/SecretProperties.cs [hunk 8]
```diff
@@ -18,14 +18,14 @@ public SecretProperties()
         /// <summary> object which contains secret parameters. </summary>
         internal SecretParameters Parameters { get; set; }
         /// <summary> The type of the Secret to create. </summary>
-        internal SecretType ParametersType
```
Comment (Yao725) [inline sdk/cdn/Azure.ResourceManager.Cdn/src/Generated/Models/SecretProperties.cs:21]: Here `Type rename` and `Single property hide` work together, which make the name weird. But considering this is internal, so I'm ok with that.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/CheckNameAvailabilityRequest.cs [hunk 9]
```diff
@@ -18,6 +18,6 @@ public CheckNameAvailabilityRequest()
         /// <summary> The name of the resource for which availability needs to be checked. </summary>
         public string Name { get; set; }
         /// <summary> The resource type. </summary>
-        public string Type { get; set; }
+        public string CheckNameAvailabilityRequestType { get; set; }
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/CheckNameAvailabilityRequest.cs:21]: Here is a common issue the you mentioned.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/ErrorResponse.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.DeviceUpdate.Models
```
Comment (Yao725) [inline sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/ErrorResponse.cs:10]: It looks like that DeviceUpdate uses the common type `ErrorResponse`, I will look at why it uses this.

**Review by Yao725 — COMMENTED** [review]
`Cdn`, `DeviceUpdate`, `EventHubs` and `ServiceBus` LGTM, thanks!

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/ErrorResponse.cs (hunk 10 above)
Comment (Yao725) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/src/Generated/Models/ErrorResponse.cs:10]: It's internal

**Review by archerzz — COMMENTED** [review]
`network`, `cosmosdb`, and `storage` look fine. Thanks.

File: sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/api/Azure.ResourceManager.DeviceUpdate.netstandard2.0.cs (hunk 1 above)
Comment (fengzhou-msft) [reply sdk/deviceupdate/Azure.ResourceManager.DeviceUpdate/api/Azure.ResourceManager.DeviceUpdate.netstandard2.0.cs]: The generator is updated to automatically rename Type to ResourceType in CheckNameAvailabilityRequest.

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 11]
```diff
@@ -3914,8 +3914,8 @@ public partial class PatchableVirtualMachineExtensionData : Azure.ResourceManage
         public string Publisher { get { throw null; } set { } }
         public object Settings { get { throw null; } set { } }
         public bool? SuppressFailures { get { throw null; } set { } }
-        public string Type { get { throw null; } set { } }
         public string TypeHandlerVersion { get { throw null; } set { } }
+        public string VirtualMachineExtensionUpdatePropertiesType { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (fengzhou-msft) [reply sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs]: updated with directive.

File: sdk/connectedvmwarevsphere/Azure.ResourceManager.ConnectedVMwarevSphere/api/Azure.ResourceManager.ConnectedVMwarevSphere.netstandard2.0.cs [hunk 12]
```diff
@@ -944,10 +944,10 @@ public partial class PatchableMachineExtensionData : Azure.ResourceManager.Conne
         public PatchableMachineExtensionData() { }
         public bool? AutoUpgradeMinorVersion { get { throw null; } set { } }
         public string ForceUpdateTag { get { throw null; } set { } }
+        public string MachineExtensionUpdatePropertiesType { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (fengzhou-msft) [reply sdk/connectedvmwarevsphere/Azure.ResourceManager.ConnectedVMwarevSphere/api/Azure.ResourceManager.ConnectedVMwarevSphere.netstandard2.0.cs]: updated with directive.

**Review by m-nash — APPROVED** [review]

**Pull request merged**